### DISCUSSION TITLE
Unable to build due to problems with "Provisioning profile".

### DISCUSSION THREAD

**Discussion by selfee-jp** [quasarframework/quasar#10434] [CLI - Capacitor / Cordova modes] [answered]
Thank you for taking a look.
I hope you can help me.

I want to do a debug build to test the actual machine.
Unable to resolve from error that device is not included.

```
mac@macnoMac-mini mobile % quasar build  -m capacitor -T ios -debug      

 Build mode........ capacitor
 Pkg quasar........ v1.15.16
 Pkg @quasar/app... v2.2.6
 Debugging......... enabled
 Publishing........ no
 Packaging mode.... xcodebuild
```

The steps I took were as follows.

1. Connected your account in Xcode.
2. Go to Manage Certificates and click on AppleDevelopment. (Role: User) 
3.Add AppleDevelopment from Manage Certificates in Team. (Role : Admin) 
4. "quasar build -m capacitor -T ios -debug --ide".
5. Build with Xcode. 6.
6. open "Certificates, Identifiers & Profiles". 
7. Download the registered Certificates and double-click them. 
8. register Identifiers.
9. register a test iPhone in Devices
10. Create Profiles, download and double-click on it.
11. "quasar build -m capacitor -T ios -debug".

> error: Provisioning profile "iOS Team Provisioning Profile: appname.app" doesn't include the currently selected device "macのMac mini" (identifier *********-****-****-*************). (in target 'App' from project 'App')
** BUILD FAILED **
 App · ⚠️  Command "xcrun" failed with exit code: 65

There is an addendum in the Podfile.
```
target 'App' do
  capacitor_pods
  # Add your Pods here
  pod 'Firebase/Core'
  pod 'Firebase/Messaging'
end
```

I've looked at the manual, etc., but I don't think I understand it.
What is the correct procedure?

It's a machine translation, so sorry for the rambling.

**Comment by ontwikkelfabriek** [comment]
@selfee-jp 

In general you have to know to develop ( and publish ) IOS apps the Apple way. This is complicated, a billion things can go wrong, if you never done this before, read ,read, read.( it took me 2 weeks of research and dealing with Apple cr@p to publish my first app to the iTunes store). I avoid it like the plague if I can. PWA's all the way.

You have to add your device to the **correct**(dev) provisioning file in the Itunes connect website:
https://itunesconnect.apple.com/login

> error: Provisioning profile "iOS Team Provisioning Profile: appname.app" 

eum `appname.app` ? You have to create a real app id that matches with the id from your registered IOS app in itunes connect. You have to set this id in the quasar.conf.js file.

btw Itunes connect is where you will manage your provisioning and development/distribution certificates.

Also I don't think you can create a **build** with the **debug** flag. Why not just use the `run` command instead?

**Reply by selfee-jp** [reply]
Android can be debugged and built.
Is it not possible for ios?

This will work.
`quasar build -m capacitor -T android -debug`

Can't build the next one?
`quasar build -m capacitor -T ios-debug`

Yes, I can use the run command to test the actual machine.
However, I don't think I can test the push notification.

The problem is not that I can run debug, but that I can't build it.

**It's a machine translation, so sorry for the rambling.**

**Reply by ontwikkelfabriek** [reply]
@selfee-jp 

I get it you want a build :-). Look at this part of my previous post:

_eum appname.app ? You have to create a real app id that **matches** with the id from your registered IOS app in **itunes connect**. You have to set this **id** in the quasar.conf.js file._

btw Please read the entire post and not just respond to one part of it.

**Reply by selfee-jp** [reply]
I'm sorry.

> eum appname.app ? You have to create a real app id that matches with the id from your registered IOS app in itunes connect. You have to set this id in the quasar.conf.js file.

appname.app is a temporary setting.It's for posting here.
It is made in accordance with quasar.conf.js.

**It's a machine translation, so sorry for the rambling.**

**Reply by ontwikkelfabriek** [reply]
@selfee-jp 

> appname.app is a temporary setting.It's for posting here. t is made in accordance with quasar.conf.js.

Ok I understand.

> error: Provisioning profile "iOS Team Provisioning Profile: appname.app" doesn't include the currently selected device "macのMac mini" (identifier ---************). (in target 'App' from project 'App')

 The problem is that your device is not added to the **correct** provisioning profile. Without access to your project and itunes connect account it's hard to guess what is wrong. What you could try is posting the itunes provisioning /certificate screenshots from itunes connect.( blur everything that is sensitive)

**Reply by selfee-jp** [reply]
@ontwikkelfabriek  Thank you.

Thank you for the screenshot suggestion.
I left the certificate as it is and re-installed the capacitor mode.

`quasar mode remove capacitor`

`quasar mode add capacitor`

`quasar build -m capacitor -T ios -debug`

> error: Signing for "App" requires a development team. Select a development team in the Signing & Capabilities editor. (in target 'App' from project 'App')
> 
> BUILD FAILED

`quasar build -m capacitor -T ios -debug —ide`
Xcode will open.
Run Product > Build.

> success!

Build it in Xcode and it will succeed.
I don't know why.

![cert](https://user-images.githubusercontent.com/9072541/130378282-0dc0ada4-e4da-4c7c-a387-488198a8dd98.png)
![Identifiers](https://user-images.githubusercontent.com/9072541/130378297-543e996b-c63d-4879-b7a6-f5f926a0533a.png)
![devices](https://user-images.githubusercontent.com/9072541/130378303-6ec21957-a16b-4641-bac3-b67e3be13e68.png)
![Provisioning](https://user-images.githubusercontent.com/9072541/130378305-b9886a7b-48ac-4a13-aa49-7dc1e4fe8ae0.png)
![Xcode](https://user-images.githubusercontent.com/9072541/130378308-d438c72a-c64f-4f29-8205-36c2ace17339.png)


**It's a machine translation, so sorry for the rambling.**

**Comment by selfee-jp** [comment, marked as answer]
I'm not sure, but it seems to have come through.
